CONTRIBUTING.md

```diff
@@ -345,6 +345,38 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
+
+Whenever possible, we adhere to semver as our minimum standards. Even before v1, we strive to not break compatibility
+without a good reason. Hence, when a change is known to cause a breaking change, it MUST be clearly marked in the
+changelog, and SHOULD include a line instructing users how to move forward.
+
+We also strive to perform breaking changes in two stages: 
+
+- when we need to remove something, we MUST mark a feature as deprecated in one version, and MAY remove it in the next one
```

Comment:
Should we also indicate deprecations in the changelog?

Reply:
Yes, and it's mentioned earlier in this section:

> Hence, when a change is known to cause a breaking change, it MUST be clearly marked in the
> changelog, and SHOULD include a line instructing users how to move forward

Reply:
I think Tigran is asking about something slightly different.  That language would apply when a feature is removed.  But before that it should be deprecated.  In many cases there will be a replacement feature added concurrent with the deprecation, and the two should be announced together in the changelog:

> - Added feature X.  This will replace feature Y, which is now deprecated.

If something is being removed with no replacement, though, we should be sure to note its deprecation and pending removal in the changelog.

Reply:
I'm sorry, I think I'm still missing something. Can you provide a suggestion to address this? I still see the current text as addressing both situations, requiring changelog entries.

Reply:
Consider scenario:
1. Feature A exists.
1. Feature B is added.
1. After a while Feature B is found to be sufficient to cover use cases where Feature A is used.
1. Feature A is marked deprecated.
1. Feature A is removed.

I think we need to have a changelog about deprecation in step 4. Deprecation _alone_ is noteworthy enough to warrant a changelog entry even if it is not happening simultaneously with some other change.

Reply:
So, this?

> - when a feature is being replaced in favor of an existing one, we MUST mark a feature as deprecated in one version, and
>  MAY remove it in the next one

Reply:
`s/the next one/a subsequent version/`

While we are presently trying to move as fast as possible and have a reasonable ability to expect users to adapt to change, these guidelines should outlast this phase, particularly when it comes to features or behavior for end users that may not be compilation breaking.  We should be prepared to have deprecation periods that last not days or weeks, but many months spanning several releases.

Reply:
My bad, I used "a subsequent one" elsewhere and should have used it here as well. Fixed now.